File: src/main/java/pl/edu/zut/mad/schedulecalendar/model/Day.java
package pl.edu.zut.mad.schedulecalendar.model;


import org.joda.time.LocalDate;

import java.util.Date;
import java.util.List;

import io.realm.RealmList;
import io.realm.RealmObject;

public class Day extends RealmObject {

    private RealmList<Lesson> lessons;
    private String time;

    public Day() {
    }

    public Day(LocalDate date, List<Lesson> lessons) {
        this.lessons = new RealmList<>();
        this.lessons.addAll(lessons);
        this.time = date.toString();
    }

    public Date getDate() {
        LocalDate date = LocalDate.parse(time);
        return date.toDate();
    }

    public List<Lesson> getLessons() {
        return lessons;
    }
}

File: src/main/java/pl/edu/zut/mad/schedulecalendar/model/Day.kt
package pl.edu.zut.mad.schedulecalendar.model

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.Ignore
import org.joda.time.LocalDate

// TODO: create api and ui model
open class Day(date: LocalDate = LocalDate.now(),
               lessons: List<Lesson> = ArrayList()) : RealmObject() {

    private var time = date.toString()

    var lessons: RealmList<Lesson> = RealmList()
        private set

    @Ignore
    var date: LocalDate? = null
        get() = LocalDate.parse(time)
        private set

    init {
        this.lessons.addAll(lessons)
    }
}
